web/src/people.ts

```diff
@@ -1533,30 +1533,23 @@ export function get_user_by_id_assert_valid(
 }
 
 function get_involved_people(message: MessageWithBooleans): DisplayRecipientUser[] {
-    let involved_people: DisplayRecipientUser[];
-
-    switch (message.type) {
-        case "stream":
-            involved_people = [
-                {
-                    full_name: message.sender_full_name,
-                    id: message.sender_id,
-                    email: message.sender_email,
-                    is_mirror_dummy: false,
-                },
-            ];
-            break;
-
-        case "private":
-            assert(
-                typeof message.display_recipient !== "string",
-                "Private messages should have list of recipients",
-            );
-            involved_people = message.display_recipient;
-            break;
-
-        default:
-            involved_people = [];
+    let involved_people: DisplayRecipientUser[] = [];
+
+    if (message.type === "stream") {
+        involved_people = [
+            {
+                full_name: message.sender_full_name,
+                id: message.sender_id,
+                email: message.sender_email,
+                is_mirror_dummy: false,
+            },
+        ];
+    } else if (message.type === "private") {
+        assert(
+            typeof message.display_recipient !== "string",
+            "Private messages should have list of recipients",
+        );
+        involved_people = message.display_recipient;
     }
```

Comment:
You can probably make this a lot nicer by just returning in each block, and just having `else { assert (message.type === "private")` for this block, rather than defining this effectively useless `involved_people` variable.

Reply:
Oh I meant to write something here earlier -- unfortunately we have some tests that rely on the default case here, and it seemed nontrivial to fix them to have more fleshed out message objects.